packages/nouns-contracts/contracts/NounsAuctionHouseV2.sol

```diff
@@ -0,0 +1,428 @@
+// SPDX-License-Identifier: GPL-3.0
+
+/// @title The Nouns DAO auction house
+
+/*********************************
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░█████████░░█████████░░░ *
+ * ░░░░░░██░░░████░░██░░░████░░░ *
+ * ░░██████░░░████████░░░████░░░ *
+ * ░░██░░██░░░████░░██░░░████░░░ *
+ * ░░██░░██░░░████░░██░░░████░░░ *
+ * ░░░░░░█████████░░█████████░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ *********************************/
+
+// LICENSE
+// NounsAuctionHouse.sol is a modified version of Zora's AuctionHouse.sol:
+// https://github.com/ourzora/auction-house/blob/54a12ec1a6cf562e49f0a4917990474b11350a2d/contracts/AuctionHouse.sol
+//
+// AuctionHouse.sol source code Copyright Zora licensed under the GPL-3.0 license.
+// With modifications by Nounders DAO.
+
+pragma solidity ^0.8.19;
+
+import { PausableUpgradeable } from '@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol';
+import { ReentrancyGuardUpgradeable } from '@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol';
+import { OwnableUpgradeable } from '@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol';
+import { IERC20 } from '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+import { INounsAuctionHouseV2 } from './interfaces/INounsAuctionHouseV2.sol';
+import { INounsToken } from './interfaces/INounsToken.sol';
+import { IWETH } from './interfaces/IWETH.sol';
+
+contract NounsAuctionHouseV2 is
+    INounsAuctionHouseV2,
+    PausableUpgradeable,
+    ReentrancyGuardUpgradeable,
+    OwnableUpgradeable
+{
+    /// @notice A hard-coded cap on time buffer to prevent accidental auction disabling if set with a very high value.
+    uint56 public constant MAX_TIME_BUFFER = 1 days;
+
+    /// @notice The Nouns ERC721 token contract
+    INounsToken public immutable nouns;
+
+    /// @notice The address of the WETH contract
+    address public immutable weth;
+
+    /// @notice The duration of a single auction
+    uint256 public immutable duration;
+
+    /// @notice The minimum price accepted in an auction
+    uint192 public reservePrice;
+
+    /// @notice The minimum amount of time left in an auction after a new bid is created
+    uint56 public timeBuffer;
+
+    /// @notice The minimum percentage difference between the last bid amount and the current bid
+    uint8 public minBidIncrementPercentage;
+
+    /// @notice The active auction
+    INounsAuctionHouseV2.AuctionV2 public auction;
+
+    /// @notice Whether this contract is paused or not
+    /// @dev Replaces the state variable from PausableUpgradeable, to bit pack this bool with `auction` and save gas
+    bool public __paused;
```

Comment:
@eladmallel I don't this is actually packs it in storage:
> Items following struct or array data always start a new storage slot.

from: https://docs.soliditylang.org/en/latest/internals/layout_in_storage.html

Reply:
you can run the build with storage layout output to see it also:

```
      {
         "astId":50,
         "contract":"contracts/NounsAuctionHouseV2.sol:NounsAuctionHouseV2",
         "label":"auction",
         "offset":0,
         "slot":"202",
         "type":"t_struct(AuctionV2)1382_storage"
      },
      {
         "astId":53,
         "contract":"contracts/NounsAuctionHouseV2.sol:NounsAuctionHouseV2",
         "label":"__paused",
         "offset":0,
         "slot":"204",
         "type":"t_bool"
      },
```

Reply:
(copy pasting comment from discord)
placing it after the `AuctionV2` struct puts it in a new storage slot.
to gain gas savings, it needs to be in the same slot as the `AuctionV2`, because only that is accessed during `settleCurrentAndCreateNewAuction`.

I'm apprehensive to add it to the struct because:
1. it's not logically part of that struct
2. we'll need to change the `auction()` view function to return a struct without paused for backwards compatibility.